frontend/src/old-pages/Clusters/JobSubmitDialog.tsx

```diff
@@ -145,55 +146,137 @@ function JobCostEstimate() {
   </>
 }
 
+interface JobFieldProps {
+  header: string;
+  description: string;
+  placeholder: string;
+  property: string;
+  disabled?: boolean;
+}
+
+function JobField({
+  header,
+  description,
+  placeholder,
+  property,
+  disabled=false
+}: JobFieldProps) {
+
+  const setStateIfNotEmpty = (detail: any) => {
+    !detail || detail === '' ? clearState([...jobPath, property]) : setState([...jobPath, property], detail);
+  }
+
+  return <>
+    <SpaceBetween direction="vertical" size="xxs" key={property}>
+      <Header 
+        variant="h2" 
+        description={description}>
+        {header}
+      </Header>
+      <FormField>
+        <Input 
+          onChange={({ detail }) => {setStateIfNotEmpty(detail.value);}}
+          value={useState([...jobPath, property])}
+          placeholder={placeholder}
+          disabled={disabled}
+        />
+      </FormField>
+    </SpaceBetween>
+  </>
+}
+
 export default function JobSubmitDialog({
   submitCallback
 }: any) {
   const { t } = useTranslation();
   const open = useState([...submitPath, 'dialog']);
   const error = useState([...submitPath, 'error']);
-  const jobPath = [...submitPath, 'job'];
   const clusterName = getState(['app', 'clusters', 'selected']);
+  const clusterPath = ['clusters', 'index', clusterName];
 
-  const job = useState(jobPath);
+  const job: {[key: string]: string} = useState(jobPath);
   const submitting = useState([...submitPath, 'pending']);
-  let jobName = useState([...jobPath, 'job-name']);
-  let chdir = useState([...jobPath, 'chdir']);
-  let nodes = useState([...jobPath, 'nodes']);
-  let ntasks = useState([...jobPath, 'ntasks']);
-  let mem = useState([...jobPath, 'mem']);
-  let command = useState([...jobPath, 'command']);
-  let wrap = useState([...jobPath, 'wrap']) || false;
+  let jobType: string = useState([...submitPath, 'job-entry']);
+
+  React.useEffect(() => jobTypeSelect('command'), []);
```

Comment:
Do we need this state to be kept globally? If not you should keep the state only inside this component (via `React.useState`). This would also allow you to initialize state via the appropriate initializer (see [here](https://reactjs.org/docs/hooks-reference.html#usestate))

This should also avoid potential UI glitches that may affect low powered devices (while the state is not initialized the UI may be different, than change after you fire the `useEffect`)